### DISCUSSION TITLE
Ubuntu Setup Stops At 64% Executing Command

### DISCUSSION THREAD

**Discussion by kiot389** [Security-Onion-Solutions/securityonion#6819] [Unsupported Versions] [answered]
I'm running so-setup on Ubuntu 18. I'm been following the install with 'tail -F soset.log.' I made a couple of adjustments to my system, but I'm at a stop where it's not continuing past executing a command. Its always at 64%. After elasticsearch-pipelines, the last line is:
`[INFO    ] Completed state [/opt/so/conf/elasticsearch/so-elasticsearch-pipelines bespin]...
[INFO    ] Running state [/usr/sbin/so-elasticsearch-templates-load] at time 23:25:54.187778
[INFO    ] Executing state cmd.run for [/usr/sbin/so-elasticsearch-templates-load]
[INFO    ] Executing command '/usr/sbin/so-elasticsearch-templates-load' in directory '/opt/so'`
I attached the logs. My AMD doesn't support CentOS7, so I hope to debug this setup
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/7841245/sosetup.log)

**Comment by weslambert** [comment]
From the log:

```
\nConnection attempt timed out.  Unable to connect to ElasticSearch.  \nPlease try: \n  -checking log(s) in /var/log/elasticsearch/\n  -running 'sudo docker ps' \n  -running 'sudo so-elastic-restart'\n\
```

It looks like the connection to Elasticsearch is timing out.  What are the resources available to this box? Also try checking the logs as mentioned in the error message for additional clues.

Thanks,
Wes

**Comment by kiot389** [comment]
Hi Wes, its strange, I check /var/log, but elasticsearch wasn't there. Its got great resources, Ryzen 5 quad-core, 16gb ddr4, 1tb m.2, asus b550 micro board. Internet access is good, only a modem out. I just kicked off set up and checking for folder creation.

**Comment by kiot389** [comment]
Elasticsearch.service & Kibana.service aren't installed. The /opt/so/elasticsearch configs are there and ready to go. Netstat is listening on my ip(not loopback):9200. I installed them from Elastic's guide, and they begin in a (dead) state. I'm gonna try a couple variations with manually installing them.

**Comment by kiot389** [comment]
I installed elasticsearch & kibana manually and got the install to 96%. After I installed them: sudo service elsaticsearch status,  says (dead). I tried starting them before the install and when it hangs at 64%, but I don't know how I got past it to 96%. It was 4am, and almost finished lol. 

@weslambert The so directory's elasticsearch.yml is configured for Security Onion, right? So, if I start the elastic service before so-setup-network it's not configured the right way. What status do you think a manual install of elasticsearch & kibana should be in before install? Is there a specific point that would explain how I got past that 64% point?

**Comment by dougburks** [comment, marked as answer]
@kiot389 Our Setup process should automatically install and configure Elasticsearch and Kibana. Please do not install them manually.

**Reply by kiot389** [reply]
@dougburks Much appreciated. It just worked. Simply installed Ubuntu 18, cloned git, ran so-setup-network. All set. Thanks.
